libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.h

```diff
@@ -42,6 +42,14 @@ typedef std::shared_ptr<WiFiEventHandlerOpaque> WiFiEventHandler;
 
 typedef void (*WiFiEventCb)(WiFiEvent_t);
 
+enum class DNSResolveType: uint8_t
+{
+    DNS_AddrType_IPv4 = 0,	// LWIP_DNS_ADDRTYPE_IPV4 = 0
+    DNS_AddrType_IPv6,		// LWIP_DNS_ADDRTYPE_IPV6 = 1
+    DNS_AddrType_IPv4_IPv6,	// LWIP_DNS_ADDRTYPE_IPV4_IPV6 = 2
+    DNS_AddrType_IPv6_IPv4	// LWIP_DNS_ADDRTYPE_IPV6_IPV4 = 3

```

Comment:
What do you think of 
```cpp
{
    DNS_AddrType_IPv4 = LWIP_DNS_ADDRTYPE_IPV4,
    DNS_AddrType_IPv6 = LWIP_DNS_ADDRTYPE_IPV6,
    DNS_AddrType_IPv4_IPv6 = LWIP_DNS_ADDRTYPE_IPV4_IPV6,
    DNS_AddrType_IPv6_IPv4 = LWIP_DNS_ADDRTYPE_IPV6_IPV4,
}
```

Reply:
Hi, this was exactly my first try but it caused problems with the CI built as some of the tests failed because of LWIP_DNS_ADDRTYPE_* isn't defined during compile time. I did put the typedef into an IF clause to have it only built if Dualstack is enabled but that caused a even more strange built problem with test 1. This was the only working solution I could find without breakting the tests.